provider/pkg/gen/typegen.go

```diff
@@ -260,7 +260,7 @@ func GVKFromRef(ref string) schema.GroupVersionKind {
 	gvk := schema.GroupVersionKind{
 		Kind:    split[len(split)-1],
 		Version: split[len(split)-2],
-		Group:   split[len(split)-3],
+		Group:   strings.Join(split[:len(split)-2], "."),
```

Comment:
This is the main behavior change. We now store `io.k8s.api.apps` as the key in the `canonicalGroup` map. Previously, this was `apps`. This is an issue for downstream consumers of this package since `io.k8s.api.apps` and `com.mytestpackage.apps` would have collided, and resulted in incorrectly generated package names.